pos_lot_selection/static/src/js/popup/LotSelectorPopup.js

```diff
@@ -0,0 +1,35 @@
+/*
+    Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
+    License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+*/
+odoo.define("pos_lot_selection.LotSelectorPopup", function (require) {
+    "use strict";
+
+    const AbstractAwaitablePopup = require("point_of_sale.AbstractAwaitablePopup");
+    const Registries = require("point_of_sale.Registries");
+    const {_lt} = require("@web/core/l10n/translation");
+
+    class LotSelectorPopup extends AbstractAwaitablePopup {
+        constructor() {
+            super(...arguments);
+            // If there's a product, get lots available related to this product
+            this.slected_items = [];
+            if (this.props.product) {
+                this.lots = this.env.pos.db.lot_by_product_id[this.props.product.id];
```

Comment:
Can we please avoid loading lots in the POS session and do and RPC call to get the list of lots from the popup instead?

Reply:
AFAIK, pos has been designed to work offline, if you call RPC in real time, wouldn't you break that feature ?

Reply:
Indeed. Moreover, now only the good ones are loaded, LG.

Reply:
Even with POS being designed to work offline, it can still work offline without having to load the lots. It's just that you'd have to enter the number manually instead of having the possibility to select, not having a lot loaded is not blocking in any way, as if you wouldn't have a product loaded and would be unable to sell it.

The possibility to select is therefore nice to have but again it's not blocking if you are offline, whereas loading the lots in the POS means that we have to sync the lots if there are many movements, to make sure they are up to date, what could easily be avoided by not loading the lots in the POS.

Finally, please have a look at pos_coupon, and you'll notice that the POS is doing a RPC request when a coupon code is entered on the point of sale, and it's impossible to apply a coupon in offline mode. I guess the reason is more or less the same than here, where you prefer to make sure that you're online to validate a coupon that having to load and sync all the time to make sure it's valid...